consensus/istanbul/core/roundstate_db.go

```diff
@@ -172,5 +226,70 @@ func (rsdb *roundStateDBImpl) GetRoundStateFor(view *istanbul.View) (RoundState,
 }
 
 func (rsdb *roundStateDBImpl) Close() error {
+	if rsdb.opts.withGarbageCollector {
+		rsdb.stopGarbageCollector()
+	}
 	return rsdb.db.Close()
 }
+
+func (rsdb *roundStateDBImpl) garbageCollectEntries() {
+	logger := rsdb.logger.New("func", "StartGarbageCollector")
+
+	oldestValidView, err := rsdb.GetOldestValidView()
+	if err != nil {
+		logger.Error("Aborting RoundStateDB GarbageCollect: Failed to fetch oldestValidView", "err", err)
+		return
+	}
+
+	logger.Debug("Prunning entries from old views", "oldestValidView", oldestValidView)
+	count, err := rsdb.deleteEntriesOlderThan(oldestValidView)
+	if err != nil {
+		logger.Error("Aborting RoundStateDB GarbageCollect: Failed to remove entries", "entries_removed", count, "err", err)
+		return
+	}
+
+	logger.Info("Finished RoundStateDB GarbageCollect", "removed_entries", count)
+}
+
+func (rsdb *roundStateDBImpl) deleteEntriesOlderThan(lastView *istanbul.View) (int, error) {
+	fromViewKey := view2Key(&istanbul.View{Sequence: common.Big0, Round: common.Big0})
+	toViewKey := view2Key(lastView)
+
+	iter := rsdb.db.NewIterator(&util.Range{Start: fromViewKey, Limit: toViewKey}, nil)
+	defer iter.Release()
+
+	counter := 0
+	for iter.Next() {
+		rawKey := iter.Key()
+		err := rsdb.db.Delete(rawKey, nil)
+		if err != nil {
+			return counter, err
+		}
+		counter++
+	}
+	return counter, nil
+}
+
+// view2Key will encode a view in binary format
+// so that the binary format maintains the sort order for the view
+func view2Key(view *istanbul.View) []byte {
+	prefix := []byte(rsKey)
+	buff := make([]byte, len(prefix)+16)
+
+	copy(buff, prefix)
+	// TODO (mcortesi) Support Seq/Round bigger than 64bits
+	binary.BigEndian.PutUint64(buff[len(prefix):], view.Sequence.Uint64())
```

Comment:
This took me a little while to picture in my head & understand this order, I think a comment that leveldb evaluates ranges of `[]byte` as one big big-endian string of bits is why we have our key structured as `[big endian sequence . big endian round ]` would be helpful to a reader in the future

We never set the sequence or round to be negative for whatever reason do we? Personally I think Uint64 is huge enough that we won't have to realistically worry about getting close to the max value

Reply:
I agree this deserves a better comment!